### PR TITLE
create-dmg: init 1.2.2

### PR DESCRIPTION
**Pull request by heywoodlh**
## Description of changes

Adds [create-dmg](https://github.com/create-dmg/create-dmg), a DMG creator for MacOS

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/cr/create-dmg/package.nix
Comment (clebs) [inline pkgs/by-name/cr/create-dmg/package.nix:17]: `InstallPhase` is not necessary in this case. This should yield the same result:
```
installFlags = [ "PREFIX=$(out)" ];
```

**Review by clebs — COMMENTED** [review]
Thanks for the contribution! 🚀 
Also your commits need to be reversed (first `maintainers` and then `create-dmg init`)

**Comment by heywoodlh** [comment]
@clebs thank you for the thoughtful review! I believe I addressed all your concerns in my most recent commit: f8ada611c660eb83d6611558081dbdbbfb50fbc8

(Sorry if the notifications I generated resolving the conversations were obnoxious)

**Comment by heywoodlh** [comment]
Also, here's how I tested functionality (it seems to work as expected):

```
/tmp/test
❯ mkdir source_folder; echo stuff > source_folder/file.txt; echo other-stuff > source_folder/file2.txt

/tmp/test
❯ create-dmg \
        --volname "Application Installer" \
        --window-pos 200 120 \
        --window-size 800 400 \
        --icon-size 100 \
        --icon "Application.app" 200 190 \
        --hide-extension "Application.app" \
        --app-drop-link 600 185 \
        "test.dmg" \
        "source_folder/"
```

And the resulting Volume in Finder with the contents:

image

File: pkgs/by-name/cr/create-dmg/package.nix
Comment (AndersonTorres) [inline pkgs/by-name/cr/create-dmg/package.nix:5]: Version should start with a digit.

File: pkgs/by-name/cr/create-dmg/package.nix
Comment (AndersonTorres) [inline pkgs/by-name/cr/create-dmg/package.nix:21]:
```suggestion
```

File: pkgs/by-name/cr/create-dmg/package.nix
Comment (AndersonTorres) [inline pkgs/by-name/cr/create-dmg/package.nix:23]:
```suggestion
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by heywoodlh** [comment]
@AndersonTorres thanks for the feedback, I believe I've addressed your feedback on the most recent commit

File: pkgs/by-name/cr/create-dmg/package.nix
Comment (AndersonTorres) [inline pkgs/by-name/cr/create-dmg/package.nix:10]: Strange. 

https://github.com/create-dmg/create-dmg/refs/tag/v1.2.2 points to nothing.

Comment (clebs) [reply pkgs/by-name/cr/create-dmg/package.nix:10]: It does not on github website but the tag is there: https://github.com/create-dmg/create-dmg/tags
If you clone the repo and do `git checkout refs/tags/v1.2.2` it works.

File: pkgs/by-name/cr/create-dmg/package.nix
Comment (clebs) [reply pkgs/by-name/cr/create-dmg/package.nix]: When changging this, also change the revision on `fetchFromGitHub` to:
`rev = "refs/tags/v${finalAttrs.version}";`

File: pkgs/by-name/cr/create-dmg/package.nix
Comment (AndersonTorres) [reply pkgs/by-name/cr/create-dmg/package.nix:10]: So it lacks the `v` before the number.

Comment (clebs) [reply pkgs/by-name/cr/create-dmg/package.nix:10]: Not sure what you mean, the b is there in the checkout.
Were you maybe commenting on my other response?

Comment (AndersonTorres) [reply pkgs/by-name/cr/create-dmg/package.nix:10]: > Not sure what you mean, the b is there in the checkout.

It should be 

`rev = "refs/tags/v${finalAttrs.version}";`

instead of

`rev = "refs/tags/${finalAttrs.version}";`

Comment (AndersonTorres) [reply pkgs/by-name/cr/create-dmg/package.nix:10]: The intention is not overriding version per se, but not requiring rec.

Comment (clebs) [reply pkgs/by-name/cr/create-dmg/package.nix:10]: Yes, that is what o said needed to be changed in the first comment. That of the v is removed from the version attribute, it needs to be added to ref

Comment (AndersonTorres) [reply pkgs/by-name/cr/create-dmg/package.nix:10]: https://nix.dev/guides/best-practices.html#recursive-attribute-set-rec

Comment (AndersonTorres) [inline pkgs/by-name/cr/create-dmg/package.nix:10]:
```suggestion
    rev = "refs/tags/v${finalAttrs.version}";
```

Comment (heywoodlh) [reply pkgs/by-name/cr/create-dmg/package.nix]: Resolved with most recent commit

Comment (heywoodlh) [reply pkgs/by-name/cr/create-dmg/package.nix]: Resolved on the most recent commit, please let me know if this is still an issue!

File: pkgs/by-name/cr/create-dmg/package.nix
Comment (heywoodlh) [reply pkgs/by-name/cr/create-dmg/package.nix]: I didn't even know `stdenvNoCC` was a thing, thanks for the pointer! Switched to it on the most recent commit.

**Review by clebs — APPROVED** [review]
LGTM!

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Reword the commit as 

```
create-dmg: init at 1.2.2
```

**Comment by heywoodlh** [comment]
@AndersonTorres done

Out of curiosity and for my future reference, is the phrasing of the commit _that important_ before being merged into master? If so, why?

**Comment by AndersonTorres** [comment]
1. This is pontificated at Contributing docs: https://github.com/NixOS/nixpkgs/blob/master/pkgs%2FREADME.md

2. The formatting allows grepping the commit logs more easily, in a similar fashion to the "conventional commits" standard (which I think it is ugly).

[XKCD 1296](https://xkcd.com/1296/)

File: pkgs/by-name/cr/create-dmg/package.nix
Comment (afh) [inline pkgs/by-name/cr/create-dmg/package.nix:19]: Please remove the indefinite article from the beginning of the description as recommended in [section "Meta attributes" in `pkgs/README.md`](https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes)
```suggestion
    description = "Shell script to build fancy DMGs";
```

**Review by afh — CHANGES_REQUESTED** [review]
Thanks for this, @heywoodlh, much appreciated. Please find below a minor suggestion for improvement.

File: pkgs/by-name/cr/create-dmg/package.nix
Comment (heywoodlh) [reply pkgs/by-name/cr/create-dmg/package.nix]: Added on most recent commit, thanks!

Comment (sarahec) [reply pkgs/by-name/cr/create-dmg/package.nix]: @afh I can confirm that change. Can we get an approval?

**Comment by afh** [comment]
Something seems off with the checks, not sure if the following is going to work to re-trigger the checks…

@ofborg test

File: pkgs/by-name/cr/create-dmg/package.nix
Comment (afh) [reply pkgs/by-name/cr/create-dmg/package.nix]: The changes look good to me, yet having all checks pass would be helpful, @sarahec.

**Comment by sarahec** [comment]
@afh the merge check times out two weeks ago. Do we ask @heywoodlh to rebase atop master and force push, knowing it'll lose the one approval (@clebs)?

**Comment by AndersonTorres** [comment]
@ofborg build create-dmg

**Comment by heywoodlh** [comment]
Let me know what you need from me and I'll be happy to make those changes

**Comment by afh** [comment]
This PRs branch is ~12k commits behind `master`, possibly that is the reason why checks time out. @heywoodlh, could you rebase this PRs branch onto master, please?

**Comment by DontEatOreo** [comment]
@heywoodlh Are you still interested in pursuing this PR?

**Comment by heywoodlh** [comment]
Rebased onto master, let me know if there's any other follow up that is needed.

**Comment by afh** [comment]
Thanks for rebase, @heywoodlh, mind formatting the package using:
`nix run nixpkgs#nixfmt-rfc-style -- pkgs/by-name/cr/create-dmg/package.nix`, please? This should address the failed nixfmt check..

**Comment by heywoodlh** [comment]
@afh done! Also, added `meta.changelog` in there while I was at it.

**Comment by afh** [comment]
Result of `nixpkgs-review pr 319628` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>create-dmg</li>
  </ul>
</details>

**Review by afh — APPROVED** [review]
Thanks for the latest updates on this PR, much appreciated 👍

**Comment by DontEatOreo** [comment]
Result of `nixpkgs-review pr 319628` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>create-dmg</li>
  </ul>
</details>

**Comment by DontEatOreo** [comment]
Result of `nixpkgs-review pr 319628` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>create-dmg</li>
  </ul>
</details>

**Review by 4evy — APPROVED** [review]

**Review by AndersonTorres — APPROVED** [review]

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 319628` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Comment by AndersonTorres** [comment]
> Result of `nixpkgs-review pr 319628` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

Not conclusive, since it is Darwin-only.

**Comment by JohnRTitor** [comment]
Yeah, tried on the nix-community darwin builder and it succeeded.
```log
johnrtitor@darwin01 ~ % nix build "github:nixos/nixpkgs/refs/pull/319628/merge#create-dmg" --print-out-paths
/nix/store/ajjyja98l3xa05w6rvrzi14arj23g4sd-create-dmg-1.2.2
nix build "github:nixos/nixpkgs/refs/pull/319628/merge#create-dmg"   1.70s user 12.60s system 60% cpu 23.516 total
johnrtitor@darwin01 ~ % ./result/bin/create-dmg
Not enough arguments. Run 'create-dmg --help' for help.
1 johnrtitor@darwin01 ~ % ./result/bin/create-dmg --help
create-dmg 1.2.1

Creates a fancy DMG file.

Usage:  create-dmg [options] <output_name.dmg> <source_folder>

All contents of <source_folder> will be copied into the disk image.

```

**Review by JohnRTitor — APPROVED** [review]

**Pull request merged**
